app/graphql/queries/user.rb

```diff
@@ -0,0 +1,26 @@
+# frozen_string_literal: true
+
+module Queries
+  class User < Queries::BaseQuery
+    description "get user by argument"
+    type Types::Models::UserType, null: false
+    argument :id, Integer, required: false
+    argument :login, String, required: false
+    argument :email, String, required: false
+    argument :name, String, required: false
+
+    def resolve(args)
+      return {} unless args
+
+      if args.key?(:id)
```

Comment:
Consider replacing these with a `case` statement or a series of `return`

Reply:
I couldn't figure out a way to phrase this with `case`, please let me know if something jumps out. It's intended to be a versatile query that can be used with one arg only.

The processing advantage of `case` as i understand it would be that the case predicate is only evaluated once, and then it's compared to the `when`s. 

But here it's looking for a different key in a small array each time, and i couldn't think of a way to process the `args` in a way that the `when`s wouldn't just be looking in some other derived array.